src/components/Reports/FinancialAccountsReport/AccountTransactions/AccountTransactions.tsx

```diff
@@ -39,44 +50,48 @@ export const AccountTransactions: React.FC = () => {
   const { query } = useRouter();
   const { t } = useTranslation();
   const locale = useLocale();
+  const { accountListId, financialAccountId, setPanelOpen } = useContext(
+    FinancialAccountContext,
+  ) as FinancialAccountType;
+
   const {
-    accountListId,
-    financialAccountId,
-    activeFilters,
+    activeFilters = {},
     setActiveFilters,
-    hasActiveFilters,
-    searchTerm,
-    setPanelOpen,
-  } = useContext(FinancialAccountContext) as FinancialAccountType;
+    isFiltered,
+    searchTerm = '',
+  } = useUrlFilters<FinancialAccountsFilters>();
+
   const { appName } = useGetAppSettings();
 
   useEffect(() => {
-    // On loading the transactions page, open the filters panel and reset the active filters.
-    // We need to reset the active filters to ensure the date range is set to the date range in the URL if it exists.
-    const urlFilters =
-      query?.filters && JSON.parse(decodeURI(query.filters as string));
     setPanelOpen(Panel.Filters);
-    setActiveFilters(urlFilters ?? {});
+
     return () => {
       setPanelOpen(null);
       setActiveFilters({});
     };
-  }, [query?.filters]);
+  }, []);
 
   const defaultDateRange = useMemo(() => formatDateRange(), []);
   const defaultStartDate = defaultDateRange.split('..')[0];
   const defaultEndDate = defaultDateRange.split('..')[1];
 
   useEffect(() => {
-    if (!hasActiveFilters && !query?.filters) {
+    if (!isFiltered && !query?.filters) {
```

Comment:
And you can remove `query?.filters` from the dependency array.

```suggestion
    if (!isFiltered) {
```